File: Linked List/61. Rotate List/rotate.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution
{
public:
    ListNode *rotateRight(ListNode *head, int k)
    {
        ListNode *tail = head;
        ListNode *curr = head;
        ListNode *temp;

        if (!head || !head->next)
        {
            return head;
        }

        int list_len = 1;
        while (tail->next)
        {
            tail = tail->next;
            list_len++;
        }

        printf("%d", list_len);

        int iter = list_len - (k % list_len);

        for (int i = 0; i < iter; i++)
        {
            temp = head;
            head = head->next;
            temp->next = nullptr;
            tail->next = temp;
            tail = tail->next;
        }
        return head;
    }
};

File: Linked List/61. Rotate List/rotate.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        tail = head
        curr = head
        temp = None

        if not head or not head.next:
            return head

        list_len = 1
        while tail.next:
            tail = tail.next
            list_len += 1

        iter = list_len - (k % list_len)

        for i in range(iter):
            temp = head
            head = head.next
            temp.next = None
            tail.next = temp
            tail = tail.next

        return head
